include/picongpu/plugins/adios/ADIOSWriter.hpp

```diff
@@ -556,23 +558,48 @@ class ADIOSWriter : public IIOBackend
         auto fieldsOffsetDims = params->fieldsOffsetDims;
 
         /* Patch for non-domain-bound fields
-         * This is an ugly fix to allow output of reduced 1d PML buffers,
-         * that are the same size on each domain.
-         * This code is to be replaced with the openPMD output plugin soon.
+         * This is an ugly fix to allow output of reduced 1d PML buffers
          */
         if( !traits::IsFieldDomainBound< T_Field >::value )
         {
             DataConnector &dc = Environment<>::get().DataConnector();
             auto field = dc.get< T_Field >( T_Field::getName() );
             fieldsSizeDims = precisionCast< uint64_t >( field->getGridLayout().getDataSpaceWithoutGuarding() );
             dc.releaseData( T_Field::getName() );
-            auto const & gridController = Environment<simDim>::get().GridController();
-            auto const numRanks = gridController.getGlobalSize();
-            auto const rank = gridController.getGlobalRank();
+            auto & gridController = Environment<simDim>::get().GridController();
+            auto const numRanks = uint64_t{ gridController.getGlobalSize() };
+            auto const rank = uint64_t{ gridController.getGlobalRank() };
```

Comment:
Only a note: @sbastrakov and me discussed offline that the rank is not the best unique id as marker to load data. The reason is that we can not guarantee that the rank is deterministic.

Reply:
We need to introduce a unique reproducible ID in pmacc. This should not be part of this PR and will break the compatibility of checkpoints written with this PR and later checkpoints.

Reply:
Discussed offline, I will change MPI rank to the rank based on domain position. There is already `gridController.getScalarPosition()` for getting it.

Reply:
Done.